x-pack/plugins/elastic_assistant/server/routes/helpers.ts

```diff
@@ -442,12 +442,15 @@ export const langChainExecute = async ({
     executorParams
   );
 
+  const isEnabledKnowledgeBase = (await kbDataClient?.isModelDeployed()) ?? false;
```

Comment:
So slight difference here between client telemetry and server telemetry in how we're determining `isEnabledKnowledgeBase`. The client check includes if the ESQL docs have been loaded, where these server checks are only ensuring ELSER is deployed, which could have happened separately from interactions with the assistant. If we want these to match, this is how you can check if the ESQL docs are present (as the get_kb_status route does):

https://github.com/elastic/kibana/blob/88ca1f52bf789179354f01f045f4f5d4d4417e47/x-pack/plugins/elastic_assistant/server/routes/knowledge_base/get_knowledge_base_status.ts#L100-L107

Note: that request will fail if the model is not deployed since it's needed for retrieval, so can just check if deployed beforehand and exit early to avoid that request (which now looking at it that check could arguable live within `getKnowledgeBaseDocumentEntries()` itself).